docs/rfcs/batch-service-pubsub-improvement.md

```diff
@@ -0,0 +1,148 @@
+# Summary
+
+Currently, we are calling the batch service to process the events from the PubSub every 1 minute. But it would be better
+if we can process the events as soon as possible.
+
+To reduce the latency of processing the events and let end users receive Slack notifications more quickly, we will make
+improvements to the batch service event processing job.
+
+## Implementation
+
+Basically, we will create a new puller to process the events from the PubSub for each topic. The new architecture should
+be able to add new topics easily, because we will have more topics in the future.
+
+### Architecture
+
+The multi pub-sub architecture will have the following components:
+
+- **Configuration**: The configuration for the subscriber, which includes the topic and the subscription name.
+- **Subscriber**: The subscriber that will pull the messages from the PubSub.
+- **Processor**: The processor that will process the messages from the PubSub.
+
+We will create a multi pub-sub Golang struct to handle the configuration, subscriber, and processor. The multi pub-sub
+will have the following methods:
+
+- **NewMultiPubSub**: The constructor for the multi pub-sub.
+- **Start**: The method to start the multi pub-sub.
+- **Stop**: The method to stop the multi pub-sub.
```

Comment:
When will the stop be called? When is the batch server's context done?

Reply:
Yes, the `Stop` function will be called when the server context is done or receives an exit signal.